enhancements/node-tuning/pao-in-nto.md

```diff
@@ -0,0 +1,258 @@
+
+[cnf-tests](https://github.com/openshift-kni/cnf-features-deploy/) are NOT part of this move either, the Telco team will keep maintaining the validation suite in OpenShift KNI.
+
+
+#### CI
+
+PAO contains both unit and functional tests implemented using Gingko.
+
+Integration with NTO is currently TBD, please advise if anything special is needed.
+
+One of the test suites (4_latency) implements an opinionated latency testing suite to get comparable results from customers and to prevent configuration / command line invocation mistakes. Keeping it in PAO/NTO or moving it into cnf-tests is TBD.
+
+#### Build pipeline
+
+PAO would inherit the NTO pipeline and all the images will be released via ART.
+
+#### Releases
+
+PAO should be shipped as part of core OCP. ART again. PAO already obeys the OCP github PR rules and uses the same PR bot so nothing changes in the lifecycle.
+
+#### API stability
+
+The `PerformanceProfile` CR uses an `performance.openshift.io/v2` API group which marks it as stable interface that needs to be maintained for about a year after official deprecation. We do not expect any change to the API as part of this proposal.
+
+#### Bugs
+
+PAO bugs are currently tracked in The OpenShift Bugzilla component [Performance Addon Operator](https://bugzilla.redhat.com/buglist.cgi?component=Performance%20Addon%20Operator&list_id=12053428&product=OpenShift%20Container%20Platform)
+
+This should be kept as-is, the Telco team will own all the bugs. We might need to move the component under NTO or rename it.
+
+The bugzilla component is currently not part of OCP metrics and we can include it. TBD
+
+
+### User Stories
+
+Both the NTO and PAO functionality should be kept intact and no regressions are allowed. The user should not notice anything except the fact that OLM is no longer needed to enable the tuning functionality.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions [optional]
+
+### Upgrade
+
+- PAO has been installed via OLM since 4.4. Once it moves to core we will have to figure out the upgrade procedure, because OLM will not remove the optional operator by itself.
+
+  One of the options is to keep using the same leadership election method and make sure the newest version always wins. The OLM operator will then stay on the cluster (can be release noted) as an inactive component.
```

Comment:
I do not think the leader election supports such conditioning. One of the options is to remove the relevant OLM stuff via the performance-addon controller and leave the CRDs without the change.

Reply:
What are the relevant OLM stuff here ?  if we remove the catalog source/csvs/subscriptions  it will probably remove the pao-olm-operator and consequently its owned CRs no ?    
This brings me to another thought, what will happen to the previous version Maching config/tuned generated artifacts.

Reply:
CSV, subscription, operator group. It will not remove the CRD, so all CR's should be untouched.

Reply:
So are we aiming for a single version finalizer here ?

Reply:
Added the OLM artifacts removal as an option to this doc.